frontend/src/Account.js

```diff
@@ -15,57 +17,120 @@ function Account() {
     useEffect(() => {
-        if (isAuthenticated && account !== undefined) {
-            axios
-                .get("/api/nft/listings", {
+        if (isAuthenticated) {
+            axios.get("/api/nft/listings", 
+                {
                     params: {
                         available: true,
                         publicAddress: account,
                     },
                 })
                 .then((res) => {
                     console.log(res.data);
+                    var data = res.data;
+                    for (var i = 0; i < data.length; i++) {
+                        if (data[i].imageUrl === undefined || data[i].imageUrl === null) {
+                            data[i].imageUrl = logo;
+                        }
+                    }
+                    setListings(data);
                 });
         }
 
-    }, [account]);
+    }, [isAuthenticated]);
 
-    async function submit() {
+    async function edit() {
+        const [ publicAddress, signature ] = await authenticateAction(navigate);
         
-        // if there is more than 1 account linked, then the user is authorized 
-        await window.ethereum.request({ method: 'eth_requestAccounts' })
-            .then((accounts) => {
-                if (accounts.length > 0) {
-                    setAuthenticated(true);
-                    setAccount(accounts[0]);
-                    console.log(account);
-                    //navigate("/")
-                }
-            })
-            .catch((error) => {
-                if (error.code === 4001) {
-                    // EIP-1193 userRejectedRequest error
-                    console.log('Please connect to MetaMask.');
-                    setAuthenticated(false);
-                } else {
-                    console.error(error);
+        axios.put('/api/nft/listing', { 
+          blockchain: 'polygon',
+          contractAddress: '0x622d8fea4603ba9edaf1084b407052d8b0a9bed7',
+          tokenID: '1000633',
+          publicAddress: '0x96a16f15Ea9204b3742156af19649DBfdAFd7B16'
+        }, {
+          params: {
+            publicAddress: publicAddress,
+            signature: signature
+          }
+        })
+        .then((res) => {
+          console.log(res.data)
+        });
+      }
+
+    async function remove(id) {
+        
+        var answer = window.confirm ("Would you like to delete this listing?");
+        if (answer) {
+            const [ publicAddress, signature ] = await authenticateAction(navigate);
+            axios.delete('/api/nft/listing',  {
+                params: {
+                  publicAddress: publicAddress,
+                  listingID: id,
+                  signature: signature,
                 }
-            });
-        //const [publicAddress, signature] = await authenticateAction(navigate);
-    }
+              })
+              .then((res) => {
+                console.log(res.data)
+                alert ('Your listing has been deleted.')
```

Comment:
Is there a better way to implement this?

Reply:
This is the simplest implementation